oqsbuilder/oqsbuildfile.yml

```diff
@@ -1,106 +1,548 @@
+        cupqc_ml-kem-512_cuda:
+          upstream: cupqc
+          param: ml_kem_512
+          arch: cuda
+          copies:
+            cupqc_ml-kem.cu: cuda/ml-kem-512/cupqc_ml-kem.cu
+          enable_by: OQS_ENABLE_KEM_ml_kem_512_cuda
+          link_libs:
+            private: [cupqc-pk_static]
+          cuda_arch: OFF
+          compile_opts:
+            private: ["$<$<COMPILE_LANGUAGE:CUDA>:-rdc=true -dlto -arch=compute_70>"]
+          keypair: "cupqc_ml_kem_512_keypair"
+          enc: "cupqc_ml_kem_512_enc"
+          dec: "cupqc_ml_kem_512_dec"
+        cupqc_ml-kem-768_cuda:
+          upstream: cupqc
+          param: ml_kem_768
+          arch: cuda
+          copies:
+            cupqc_ml-kem.cu: cuda/ml-kem-768/cupqc_ml-kem.cu
+          enable_by: OQS_ENABLE_KEM_ml_kem_768_cuda
+          link_libs:
+            private: [cupqc-pk_static]
+          cuda_arch: OFF
+          compile_opts:
+            private: ["$<$<COMPILE_LANGUAGE:CUDA>:-rdc=true -dlto -arch=compute_70>"]
+          keypair: "cupqc_ml_kem_768_keypair"
+          enc: "cupqc_ml_kem_768_enc"
+          dec: "cupqc_ml_kem_768_dec"
+        icicle_ml-kem-1024_icicle_cuda:
+          upstream: icicle
+          param: ml_kem_1024
+          arch: icicle_cuda
+          copies:
+            icicle_ml-kem.cpp: icicle_cuda/ml-kem-1024/icicle_ml-kem.cpp
+          enable_by: OQS_ENABLE_KEM_ml_kem_1024_icicle_cuda
+          link_libs:
+            private: [icicle::icicle_pqc_package]
+          keypair: "icicle_ml_kem_1024_keypair"
+          enc: "icicle_ml_kem_1024_enc"
+          enc_derand: "icicle_ml_kem_1024_enc_derand"
+          dec: "icicle_ml_kem_1024_dec"
+        icicle_ml-kem-512_icicle_cuda:
+          upstream: icicle
+          param: ml_kem_512
+          arch: icicle_cuda
+          copies:
+            icicle_ml-kem.cpp: icicle_cuda/ml-kem-512/icicle_ml-kem.cpp
+          enable_by: OQS_ENABLE_KEM_ml_kem_512_icicle_cuda
+          link_libs:
+            private: [icicle::icicle_pqc_package]
+          keypair: "icicle_ml_kem_512_keypair"
+          enc: "icicle_ml_kem_512_enc"
+          enc_derand: "icicle_ml_kem_512_enc_derand"
+          dec: "icicle_ml_kem_512_dec"
+        icicle_ml-kem-768_icicle_cuda:
+          upstream: icicle
+          param: ml_kem_768
+          arch: icicle_cuda
+          copies:
+            icicle_ml-kem.cpp: icicle_cuda/ml-kem-768/icicle_ml-kem.cpp
+          enable_by: OQS_ENABLE_KEM_ml_kem_768_icicle_cuda
+          link_libs:
+            private: [icicle::icicle_pqc_package]
+          keypair: "icicle_ml_kem_768_keypair"
+          enc: "icicle_ml_kem_768_enc"
+          enc_derand: "icicle_ml_kem_768_enc_derand"
+          dec: "icicle_ml_kem_768_dec"
+
+architectures:
+  portable:
+    enable_by:
+  x86_64:
+    enable_by:
+  aarch64:
+    enable_by:
+  cuda:
+    enable_by: OQS_USE_CUPQC
+  icicle_cuda:
+    enable_by: OQS_USE_ICICLE
+
+copies:
+  mlkem-native_aarch64:
```

Comment:
I worry that specifying each upstream file explicitly will significantly bloat this file and shift maintenance burden to liboqs whenever upstreams add/remove/rename files. For the mlkem-native case, the upstream repository currently maintains this in their "integrations" folder, which allows them to test the liboqs integration in their CI.